src/sentry/rules/conditions/event_frequency.py

```diff
@@ -261,15 +261,24 @@ def disable_consistent_snuba_mode(
         return option_override_cm
 
     def get_comparison_start_end(
-        self, interval: timedelta, duration: timedelta
+        self, end: datetime, duration: timedelta, comparison_interval: timedelta | None = None
     ) -> tuple[datetime, datetime]:
         """
-        Calculate the start and end times for the query. `interval` is only used for EventFrequencyPercentCondition
-        as the '5 minutes' in The issue affects more than 100 percent of sessions in 5 minutes, otherwise it's the current time.
-        `duration` is the time frame in which the condition is measuring counts, e.g. the '10 minutes' in
-        "The issue is seen more than 100 times in 10 minutes"
+        Calculate the start and end times for the query.
+        Take these example conditions that generate queries:
+            '# of issues in 1 day is {}'
+            "# of issues is {}% higher in 1 day compared to 12 hours ago"
+
+        "duration" is the period of time we're querying over, in this case 1 day
+        for both queries.
+        "comparison_interval" only applies for percent comparison queries (the
+        second condition) and is the time diff between the two queries we derive a
+        percentage from, in this case 12 hours. The two queries would be:
+            1. now -> 24 hours ago
+            2. 12 hours ago -> 36 hours ago
         """
-        end = timezone.now() - interval
+        if comparison_interval:
+            end = end - comparison_interval
```

Comment:
If we're already passing in the `end` - why do we pass in the offset?
Can we just calculate the new `end` when invoking this helper?

Reply:
We could but I liked having this function handle all the time calculations instead of doing it in `get_rate_bulk` and `get_rate`